server/clients/stellar/transfer.go

```diff
@@ -146,43 +147,48 @@ func (c *Client) GetTfchainBridgeAddress() (string, error) {
 	}
 }
 
-func (c *Client) AwaitTransactionWithMemo(account string, memo string, timeout int) error {
+func (c *Client) AwaitTransactionWithMemo(ctx context.Context, account string, memo string, timeout int) error {
 	memo = strings.TrimPrefix(memo, "0x")
 	for i := 0; i < int(timeout); i++ {
-		transactionRequest := horizonclient.TransactionRequest{
-			ForAccount: account,
-			Order:      horizonclient.OrderDesc,
-		}
-		txs, err := c.horizon.Transactions(transactionRequest)
-		if err != nil {
-			return err
-		}
-		for _, tx := range txs.Embedded.Records {
-			decodedMemo, err := base64.StdEncoding.DecodeString(tx.Memo)
-			if err == nil {
-				hexDecodedMemo := hex.EncodeToString(decodedMemo)
-				if hexDecodedMemo == memo {
-					return nil
+		select {
+		case <-time.After(1 * time.Second):
+
+			transactionRequest := horizonclient.TransactionRequest{
+				ForAccount: account,
+				Order:      horizonclient.OrderDesc,
+			}
+			txs, err := c.horizon.Transactions(transactionRequest)
+			if err != nil {
+				return err
+			}
+			for _, tx := range txs.Embedded.Records {
+				decodedMemo, err := base64.StdEncoding.DecodeString(tx.Memo)
+				if err == nil {
+					hexDecodedMemo := hex.EncodeToString(decodedMemo)
+					if hexDecodedMemo == memo {
+						return nil
+					}
 				}
 			}
+		case <-ctx.Done():
+			return nil
```

Comment:
return ctx.Err()

This way the calling code will not continue if it checks the error. Normally cancelled contexts are not checked after every call where a context is passed but the error is.